go/integTest/shared_commands_test.go

```diff
@@ -1687,6 +1687,56 @@ func (suite *GlideTestSuite) TestSPop_LastMember() {
 	})
 }
 
+func (suite *GlideTestSuite) TestSUnion() {
+	suite.runWithDefaultClients(func(client api.BaseClient) {
+		key1 := "{key}-1-" + uuid.NewString()
+		key2 := "{key}-2-" + uuid.NewString()
+		key3 := "{key}-3-" + uuid.NewString()
+		nonSetKey := uuid.NewString()
+		memberList1 := []string{"a", "b", "c"}
+		memberList2 := []string{"b", "c", "d", "e"}
+
+		res1, err := client.SAdd(key1, memberList1)
+		assert.Nil(suite.T(), err)
+		assert.Equal(suite.T(), int64(3), res1.Value())
+		assert.False(suite.T(), res1.IsNil())
+
+		res2, err := client.SAdd(key2, memberList2)
+		assert.Nil(suite.T(), err)
+		assert.Equal(suite.T(), int64(4), res2.Value())
+		assert.False(suite.T(), res2.IsNil())
+
+		res3, err := client.SUnion([]string{key1, key2})
+		assert.Nil(suite.T(), err)
+		assert.Contains(suite.T(), res3, api.CreateStringResult("a"))
+		assert.Contains(suite.T(), res3, api.CreateStringResult("b"))
+		assert.Contains(suite.T(), res3, api.CreateStringResult("c"))
+		assert.Contains(suite.T(), res3, api.CreateStringResult("d"))
+		assert.Contains(suite.T(), res3, api.CreateStringResult("e"))
```

Comment:
doesn't assert.Equal do a deep comparison? is this necessary?

Reply:
i've tried with assert.Equal, but it doesn't perform what it's supposed to like assert.Contains does

Reply:
Try `reflect.DeepEqual` (returns true/false)